src/seedu/addressbook/data/person/Printable.java

```diff
@@ -0,0 +1,19 @@
+package seedu.addressbook.data.person;
+
+public interface Printable {
+
+    String getPrintableString();
+
+    /**
+     * converts multiple Printables into a single string, separated by spaces
+     * @param printables is a varargs collection of Printables
+     * @return a string representing the arguments, in order, separated by spaces
+     */
+    static String getPrintableString(Printable... printables) {
+        String output = "";
+        for (Printable p : printables) {
```

Comment:
Java has a `StringJoiner` class that should be useful for joining strings. Check it out.

Ideally, this `getPrintableString` should be used somewhere. For example in the file [ReadOnlyPerson.java](https://github.com/nus-cs2103-AY1718S1/addressbook-level3/blob/master/src/seedu/addressbook/data/person/ReadOnlyPerson.java#L71) you can replace `getPhone()` with `getPrintableString(getPhone())`